gossip/src/cluster_info.rs

```
@@ -217,11 +217,10 @@ impl ClusterInfo {
 
     fn refresh_push_active_set(
         &self,
-        recycler: &PacketBatchRecycler,
         stakes: &HashMap<Pubkey, u64>,
         gossip_validators: Option<&HashSet<Pubkey>>,
         sender: &impl ChannelSend<PacketBatch>,
-    ) {
+    ) -> Result<(), GossipError> {
```

What is the purposed of adding error handling here if we are going to unwrap the error at the callsite? This is potentially taking a code that will silently ignore errors (which is not great but not terrible) and turning it into a code which can easily panic at runtime.

---

ahh looks like `make_gossip_packet_batch` has to now return an error which gets propagated up to `send_gossip_packets()`. but you make a good point, we shouldn't `.unwrap()` in the caller.

---

Good point, I will remove the error handling changes. More details in the comment below. :sweat_smile: